netmiko/base_connection.py

```diff
@@ -379,6 +406,12 @@ def read_until_prompt_or_pattern(self, pattern='', re_flags=0):
             combined_pattern = r"({}|{})".format(combined_pattern, pattern)
         return self._read_channel_expect(combined_pattern, re_flags=re_flags)
 
+    def serial_login(self, pri_prompt_terminator=r'#\s*$', alt_prompt_terminator=r'>\s*$',
+                     username_pattern=r"(?:[Uu]ser:|sername|ogin)", pwd_pattern=r"assword",
+                     delay_factor=1, max_loops=20):
+        self.telnet_login(pri_prompt_terminator, alt_prompt_terminator, username_pattern,
+                          pwd_pattern, delay_factor, max_loops)
+
```

Comment:
This is fine...but I probably will abstract out the common login code into a new method (i.e. the username/password) handling and then call it from both places.